subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/AdoptOpenJdkRemoteBinary.java

```diff
@@ -44,16 +46,19 @@ public AdoptOpenJdkRemoteBinary(SystemInfo systemInfo, OperatingSystem operating
     }
 
     public Optional<File> download(JavaToolchainSpec spec, File destinationFile) {
-        if (!canProvidesMatchingJdk(spec)) {
+        if (!canProvideMatchingJdk(spec)) {
             return Optional.empty();
         }
         URI source = toDownloadUri(spec);
         downloader.download(source, destinationFile);
         return Optional.of(destinationFile);
     }
 
-    private boolean canProvidesMatchingJdk(JavaToolchainSpec spec) {
-        return getLanguageVersion(spec).canCompileOrRun(8);
+    private boolean canProvideMatchingJdk(JavaToolchainSpec spec) {
+        final boolean matchesLanguageVersion = getLanguageVersion(spec).canCompileOrRun(8);
+        final JvmVendorSpec vendorSpec = spec.getVendor().get();
+        boolean matchesVendor = vendorSpec == JvmVendorSpec.ADOPTOPENJDK || vendorSpec == DefaultJvmVendorSpec.ANY;
```

Comment:
Won't you miss someone using a `JvmVendorSpec` with a match `String` that works against the AdoptOpenJDK vendor ?

Reply:
Hmmm but what would we compare it against? Given we don't have a vendor string without a JVM, we can't really compare it against anything. And just comparing it to our indicator string might result to mismatches as well (e.g. matching("inc") matches "Apple Inc" for toolchain matching but our indicator string only knows about "apple" so we can't determine it here). Any thoughts? Just compare this with the indicator string? Or leave it out so it only works if you use the known vendor